src/bun.js/api/server/StaticRoute.zig

```diff
@@ -141,14 +141,28 @@ pub fn fromJS(globalThis: *jsc.JSGlobalObject, argument: jsc.JSValue) bun.JSErro
         var headers: Headers = if (response.getInitHeaders()) |h|
             Headers.from(h, bun.default_allocator, .{
                 .body = &blob,
-            }) catch {
+            }) catch |err| {
                 blob.detach();
-                globalThis.throwOutOfMemory();
-                return error.JSError;
+                switch (err) {
+                    error.OutOfMemory => {
+                        globalThis.throwOutOfMemory();
+                        return err;
+                    },
+                    else => return err,
+                }
             }
         else
-            .{
-                .allocator = bun.default_allocator,
+            Headers.from(null, bun.default_allocator, .{
+                .body = &blob,
+            }) catch |err| {
+                blob.detach();
+                switch (err) {
+                    error.OutOfMemory => {
+                        globalThis.throwOutOfMemory();
+                        return err;
+                    },
+                    else => return err,
+                }
             };
```

Comment:
This duplication is ridiculous. Why can't it just be one call? `Headers.from(response.getInitHeaders(), ...)`